examples/telescope/components/react/proposal-modal.tsx

```diff
@@ -0,0 +1,647 @@
+  votes: Votes | undefined;
+  updateVotes: () => void;
+}) => {
+  const [option, setOption] = useState<number>();
+  const [showMore, setShowMore] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const {
+    isOpen: isVoteOptionsOpen,
+    onOpen: onVoteOptionsOpen,
+    onClose: onVoteOptionsClose,
+  } = useDisclosure();
+  const toast = useToast();
+  const { colorMode } = useColorMode();
+
+  const coin = getCoin(chainName);
+  const exponent = getExponent(chainName);
+  const { address, getSigningStargateClient } = useChain(chainName);
+
+  const chartData = [
+    {
+      title: 'YES',
+      value: Number(proposal.finalTallyResult?.yes),
+      color: VoteColor.YES,
+    },
+    {
+      title: 'NO',
+      value: Number(proposal.finalTallyResult?.no),
+      color: VoteColor.NO,
+    },
+    {
+      title: 'NWV',
+      value: Number(proposal.finalTallyResult?.noWithVeto),
+      color: VoteColor.NWV,
+    },
+    {
+      title: 'ABSTAIN',
+      value: Number(proposal.finalTallyResult?.abstain),
+      color: VoteColor.ABSTAIN,
+    },
+  ];
+
+  const emptyChartData = [
+    {
+      title: 'NO VOTES YET',
+      value: 100,
+      color: VoteColor.ABSTAIN,
+    },
+  ];
+
+  const totalVotes = useMemo(() => {
+    if (!proposal.finalTallyResult) return 0;
+    const total = Object.values(proposal.finalTallyResult).reduce(
+      (prev, cur) => prev + Number(cur),
+      0
+    );
+    return total ? total : 0;
+  }, [proposal]);
+
+  const isVoted = !!(votes && votes[proposal.proposalId.toString()]);
+
+  const isDepositPeriod = proposal.status === ProposalStatus_pb.DEPOSIT_PERIOD;
+
+  const isVotingPeriod = proposal.status === ProposalStatus_pb.VOTING_PERIOD;
+
+  const turnout = totalVotes / Number(bondedTokens);
+
+  const minStakedTokens =
+    quorum && exponentiate(quorum * Number(bondedTokens), -exponent).toFixed(6);
+
+  const description =
+    decodeUint8Arr(proposal.content?.value)
+      .match(/(?<=\u0012).*/s)?.[0]
+      .slice(2) || '';
+
+  const renderedDescription =
+    description.length > 200
+      ? showMore
+        ? description
+        : `${description.slice(0, 200)}...`
+      : description || '';
+
+  const showToast = (code: number) => {
+    toast({
+      title: `Transaction ${code === 0 ? 'successful' : 'failed'}`,
+      status: code === 0 ? 'success' : 'error',
+      duration: 3000,
+      isClosable: true,
+      position: 'top-right',
+    });
+  };
+
+  const handleConfirmClick = async () => {
+    setIsLoading(true);
+
+    const stargateClient = await getSigningStargateClient();
+
+    if (!stargateClient || !address || !option) {
+      console.error('stargateClient/address undefined or option not selected');
+      return;
+    }
+
+    const { vote: createVoteMsg } =
+      cosmos.gov.v1beta1.MessageComposer.fromPartial;
+
+    const msg = createVoteMsg({
+      option,
+      voter: address,
+      proposalId: proposal.proposalId,
```

Comment:
the `proposalId` is the `BigInt` I believe?

Reply:
yes, it's bigint here.